### PR TITLE
SFD-3376 Create postcode mapping repo

### PR DESCRIPTION
**Pull request by ArifNhs**
SFD 3376 create postcode mapping repo

### CODE REVIEW

File: build/automation/var/profile/local.mk [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+# Service variables
+
+PROFILE := local
+
+CERTIFICATE_DOMAIN := certificate/certificate
+SERVER_PORT := 8443
+VERSION := v0.0.1
+
+POSTCODE_LOCATION_DYNAMO_URL := http://host.docker.internal:8000/
+DYNAMODB_POSTCODE_LOC_MAP_TABLE := service-finder-nonprod-postcode-location-mapping
```
Comment (jope1) [inline build/automation/var/profile/local.mk:13]: Should be ' service-finder-local-postcode-location-mapping' in the local profile?

**Review by jope1 — CHANGES_REQUESTED** [review]

File: application/src/main/java/uk/nhs/digital/uec/api/constants/AuthenticationConstants.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  public static final String PASSWORD = "PASSWORD";
+  public static final String USER_PASSWORD_AUTH = "USER_PASSWORD_AUTH";
+  public static final String CLAIM_NAME = "sub";
+  public static final String ACCESS_TOKEN = "ACCESS_TOKEN";
+  public static final String REFRESH_TOKEN = "REFRESH_TOKEN";
+  public static final String COGNITO_GROUPS = "cognito:groups";
+  public static final String IDENTITY_PROVIDER_ID = "IDENTITY_PROVIDER_ID";
+  public static final String USER_HASH = "USER_HASH";
+  public static final String ROLE_PREFIX = "ROLE_";
+  public static final String POSTCODE_ROLE = "POSTCODE";
```
[the remark this answers is not in the document]
Comment (ArifNhs) [reply application/src/main/java/uk/nhs/digital/uec/api/constants/AuthenticationConstants.java:17]: I have made this changes, thanks

File: application/src/main/java/uk/nhs/digital/uec/api/domain/PostcodeMapping.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
+package uk.nhs.digital.uec.api.domain;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBAttribute;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBHashKey;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBTable;
+import lombok.Getter;
+import lombok.Setter;
+
+@DynamoDBTable(tableName = "service-finder-local-postcode-location-mapping")
```
[the remark this answers is not in the document]
Comment (ArifNhs) [reply application/src/main/java/uk/nhs/digital/uec/api/domain/PostcodeMapping.java:9]: the table name defined here is the default name but the table name will be overridden in method dynamoDBMapperConfig() in DynamoConfig class (attribute : dynamoDbTableName).

File: build/automation/var/profile/local.mk (hunk 1 above)
Comment (ArifNhs) [reply build/automation/var/profile/local.mk:13]: relevant changes made

File: README.md [hunk 4]
```diff
@@ -50,10 +50,17 @@
...
 
 ## Quick Start
 
+Run the following make targets:
+- make build
+- make start
+
+The API can be reached on this endpoint:
+
+https://localhost:8443/api/postcode
```
[the remark this answers is not in the document]
Comment (ArifNhs) [reply README.md:63]: Changes made accordingly

**Review by jope1 — DISMISSED** [review]

**Review by jope1 — APPROVED** [review]

**Pull request merged**